### PR TITLE
Validate presence of project on Staging::Workflow.

### LINKED ISSUE #7175: Missing validation for project in Staging:Workflow
From `src/api/app/models/staging/workflow.rb`:

```ruby
 parent = Project.find_or_initialize_by(name: "#{project.name}:Staging:#{letter}")
```

which is called in an after create. But it is not compulsory to have a project (at least there is not validation) which causes:

`undefined method `name' for nil:NilClass`

Solution: Add the validation and not allow to have an staging workflow without a project and fixed everything this breaks.

### PR DESCRIPTION
**Pull request by bjoernalbers**
This fixes #7175.

I've added one validation with corresponding tests.

### CODE REVIEW

**Review by danidoni — APPROVED** [review]

File: src/api/spec/models/staging/workflow_spec.rb [hunk 1]
```diff
@@ -20,6 +20,11 @@
     staging_workflow
   end
 
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:managers_group) }
+    it { is_expected.to validate_presence_of(:project) }
```
Comment (hennevogel) [inline src/api/spec/models/staging/workflow_spec.rb:25]: We usually don't test the (presence of) validations. Only if they contain logic.

**Comment by dmarcoux** [comment]
I wonder if we would need a data migration in addition to the validation. There might be staging workflows without a valid project. I didn't check...

**Comment by hennevogel** [comment]
Indeed there are, but they are not accessible anymore anyway as we route through the project. I don't think we need to do anything about them in this PR.

File: src/api/spec/models/staging/workflow_spec.rb (hunk 1 above)
Comment (vpereira) [reply src/api/spec/models/staging/workflow_spec.rb:25]: hm, i like the tests to make sure that the validation exists and wasn't disabled (i.e some disable it to debug something else, and forget to enable it back) or removed by mistake.

Comment (bjoernalbers) [reply src/api/spec/models/staging/workflow_spec.rb:25]: If it's not possible to create a staging workflow w/o associated project (via the UI) there might be no other way to test the validation.
Plus these tests are quite cheap, no need to hit the db either.

**Review by vpereira — APPROVED** [review]

**Comment by vpereira** [comment]
> Indeed there are, but they are not accessible anymore anyway as we route through the project. I don't think we need to do anything about them in this PR.

In production:

```
 > Staging::Workflow.where(project_id: nil).first
 => 7
```

**Comment by vpereira** [comment]
Thank you for your contribution @bjoernalbers!

**Pull request merged**